shell/platform/darwin/common/framework/Source/FlutterStandardCodec.mm

```diff
@@ -387,70 +395,79 @@ - (FlutterStandardTypedData*)readTypedDataOfType:(FlutterStandardDataType)type {
   UInt32 elementCount = [self readSize];
   UInt8 elementSize = elementSizeForFlutterStandardDataType(type);
   [self readAlignment:elementSize];
-  NSData* data = [self readData:elementCount * elementSize];
+  NSData* data = (NSData*)CFBridgingRelease([self readDataRef:elementCount * elementSize]);
   return [FlutterStandardTypedData typedDataWithData:data type:type];
 }
 
 - (nullable id)readValue {
-  return [self readValueOfType:[self readByte]];
+  return (__bridge id)ReadValue(self);
+}
+
+static CFTypeRef ReadValue(FlutterStandardReader* codec) {
+  return ReadValueOfType(codec, [codec readByte]);
 }
 
 - (nullable id)readValueOfType:(UInt8)type {
+  return (__bridge id)ReadValueOfType(self, type);
+}
+
+static CFTypeRef ReadValueOfType(FlutterStandardReader* codec, UInt8 type) {
   FlutterStandardField field = (FlutterStandardField)type;
   switch (field) {
     case FlutterStandardFieldNil:
       return nil;
     case FlutterStandardFieldTrue:
-      return @YES;
+      return (__bridge CFBooleanRef) @YES;
     case FlutterStandardFieldFalse:
-      return @NO;
+      return (__bridge CFBooleanRef) @NO;
     case FlutterStandardFieldInt32: {
       SInt32 value;
-      [self readBytes:&value length:4];
-      return @(value);
+      [codec readBytes:&value length:4];
+      return (__bridge CFNumberRef) @(value);
     }
     case FlutterStandardFieldInt64: {
       SInt64 value;
-      [self readBytes:&value length:8];
-      return @(value);
+      [codec readBytes:&value length:8];
+      return (__bridge CFNumberRef) @(value);
     }
     case FlutterStandardFieldFloat64: {
       Float64 value;
-      [self readAlignment:8];
-      [self readBytes:&value length:8];
-      return [NSNumber numberWithDouble:value];
+      [codec readAlignment:8];
+      [codec readBytes:&value length:8];
+      return (__bridge CFNumberRef)[NSNumber numberWithDouble:value];
     }
     case FlutterStandardFieldIntHex:
     case FlutterStandardFieldString:
-      return [self readUTF8];
+      return (__bridge CFStringRef)[codec readUTF8];
     case FlutterStandardFieldUInt8Data:
     case FlutterStandardFieldInt32Data:
     case FlutterStandardFieldInt64Data:
     case FlutterStandardFieldFloat32Data:
     case FlutterStandardFieldFloat64Data:
-      return [self readTypedDataOfType:FlutterStandardDataTypeForField(field)];
+      return (__bridge CFTypeRef)[codec readTypedDataOfType:FlutterStandardDataTypeForField(field)];
     case FlutterStandardFieldList: {
-      UInt32 length = [self readSize];
-      NSMutableArray* array = [NSMutableArray arrayWithCapacity:length];
+      UInt32 length = [codec readSize];
+      CFMutableArrayRef array = CFArrayCreateMutable(kCFAllocatorDefault, length, &kCFTypeArrayCallBacks);
```

Comment:
If you want to minimize the diff you could say:
```objc
CFMutableArrayRef array = (__bridge CFMutableArrayRef)[NSMutableArray arrayWithCapacity:length];
```

Then remove the `CFAutorelease` below.

Same thing for the dictionary below.